docs/spec/sources.md

```diff
@@ -0,0 +1,148 @@
+# Sources
+
+A **source** is any resource that generates or imports an event and relays that
+event to another resource on the cluster via CloudEvents. Sourcing events is
+critical to developing a distributed system that reacts to events.
+
+A Source:
+
+- Represents an off or on-cluster system, service or application that produces
+  events to be consumed by a sink.
+- Produces or imports CloudEvents.
+- Sends CloudEvents to the configured **sink**.
+
+In practice, sources are an abstract concept that allow us to create declarative
+configurations through the usage of Custom Resource Definitions (CRDs) extending
+Kubernetes. Those CRDs are instantiated by creating an instance of the resource.
+It is up to the implementation of the source author to understand the best way
+to realize the source application. This could be as 1:1 deployments inside of
+Kubernetes per resource, as a single multi-tenant application, or even an
+off-cluster implementation; or all combinations in-between.
+
+There are some guidelines on implementing sources to allow cluster operators and
+tools to dynamically discover and understand source installations.
+
+## Source CRDs
+
+CRDs that are to be understood as a `source` MUST be labeled:
```

Comment:
Let's expand a bit on this. It would be maybe be useful to include the motivation here, for example, so support the Source in the listing (for example kubectl get ... command). So in addition to MUST be labeled, why would be helpful. Understood is just a bit of a vague term here.

Reply:
expanded!